src/Sentry.Maui/Internal/MauiEventsBinder.cs

```diff
@@ -1,364 +1,283 @@
 using Microsoft.Extensions.Options;
-using Sentry.Extensibility;
 
 namespace Sentry.Maui.Internal;
 
-internal class MauiEventsBinder : IMauiEventsBinder
+internal class MauiEventsBinder// : IMauiEventsBinder
 {
-    private readonly IHub _hub;
     private readonly SentryMauiOptions _options;
 
-    // https://develop.sentry.dev/sdk/event-payloads/breadcrumbs/#breadcrumb-types
-    // https://github.com/getsentry/sentry/blob/master/static/app/types/breadcrumbs.tsx
-    internal const string NavigationType = "navigation";
-    internal const string SystemType = "system";
-    internal const string UserType = "user";
-    internal const string HandlersCategory = "ui.handlers";
-    internal const string LifecycleCategory =  "ui.lifecycle";
-    internal const string NavigationCategory = "navigation";
-    internal const string RenderingCategory = "ui.rendering";
-    internal const string UserActionCategory = "ui.useraction";
-
-    // This list should contain all types that we have explicitly added handlers for their events.
-    // Any elements that are not in this list will have their events discovered by reflection.
-    internal static readonly HashSet<Type> ExplicitlyHandledTypes = new()
-    {
-        typeof(BindableObject),
-        typeof(Element),
-        typeof(VisualElement),
-        typeof(Application),
-        typeof(Window),
-        typeof(Shell),
-        typeof(Page),
-        typeof(Button),
-    };
-
     public MauiEventsBinder(IHub hub, IOptions<SentryMauiOptions> options)
     {
-        _hub = hub;
         _options = options.Value;
+
+        MauiEvents.Hub = hub;
+        MauiEvents.Options = _options;
     }
 
-    public void BindApplicationEvents(Application application)
+    public void HandleApplicationEvents(Application application, bool bind = true)
```

Comment:
Where do we call this with `bind=false`? Looking at when do we unregister events. I imagine when the object we subscribed from is done with we need to do that?

Reply:
There are two places:
1. We hook in the lifecycle events in `src/Sentry.Maui/SentryMauiAppBuilderExtensions.cs` and remove ourselves on shutdown/terminate
2. The `application` has `application.DescendantRemoved += OnApplicationOnDescendantRemoved;` and we crawl through all the events we bound to and remove ourselves.